src/audio/drc/drc_generic.c

```diff
@@ -246,7 +248,7 @@ void drc_update_envelope(struct drc_state *state, const struct sof_drc_params *p
 		envelope_rate = ONE_Q20 - drc_pow_fixed(x, p->one_over_attack_frames); /* Q12.20 */
 	}
 
-	state->envelope_rate = Q_SHIFT_LEFT(envelope_rate, 20, 30);
+	state->envelope_rate = sat_int32(Q_SHIFT_LEFT((int64_t)envelope_rate, 20, 30));
```

Comment:
Why int64_t typecast is required ?
What is the  advantage of adding saturation function check here ? 
Do we have a bug in Q_SHIFT_LEFT calculation ?